src/cloud-api-adaptor/pkg/podnetwork/tunneler/config.go

```diff
@@ -0,0 +1,20 @@
+// (C) Copyright Confidential Containers Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package tunneler
+
+type TunnelerConfigurator interface {
+	Tunneler
+	Configure(*NetworkConfig, *Config) error
+}
+
+type NetworkConfig struct {
```

Comment:
What are your thoughts about making this extensible for other potential Tunnel configs? Is this a VXLAN specific implementation of NetworkConfig as it has the `VXLANConfig` as a field, if so should it be renamed to something like `VXLANNetworkConfig`. I'm guessing not as you have split out the VXLANConfig into a separate structs?

Reply:
yup, I was wondering the same. should there be a tunneling interface and the VXLAN specifics hidden behind it?

Reply:
yup, makes sense if we map cli params directly into config structs.